src/erdstall.ts

```diff
@@ -57,21 +57,36 @@ export interface Trader {
 export interface Depositor {
 	deposit(
 		assets: Assets,
-	): Promise<Stages<Promise<ethers.ContractTransaction>>>;
+	): Promise<
+		[
+			AsyncGenerator<[StageName, ethers.ContractTransaction], void, void>,
+			number,
+		]
+	>;
```

Comment:
Optimally the caller then would just use the return value using the destructuring assignment for objects, s.t. the following codechange only has to apply (given that the member names match):

```ts
		const [stages, numStages] = await conn.deposit(assets);
```
to
```ts
		const {stages, numStages} = await conn.deposit(assets);
```
The stage itself (`[StageName, ContractTransaction]`) can stay a tuple, since the type is expressive enough on its own IMHO.